src/Manager/Environment/effects/removeEnvironment.ts

```diff
@@ -23,27 +23,42 @@ import {
 } from '../environmentReducer';
 import { removeDir } from './removeDir';
 
-const removeLegacyEnvironment = (dispatch: Dispatch, toolchainDir: string) =>
-    removeDir(dispatch, path.dirname(toolchainDir));
+const removeLegacyEnvironment = (toolchainDir: string) =>
+    removeDir(path.dirname(toolchainDir));
 
-const removeNrfutilEnvironment = (dispatch: Dispatch, version: string) =>
-    Promise.all([
-        removeToolchain(version),
-        removeDir(dispatch, sdkPath(version)),
-    ]);
+const removeNrfutilEnvironment = (version: string) => {
+    removeDir(sdkPath(version));
+    try {
+        removeToolchain(version);
+    } catch (err) {
+        throw new Error(
+            `Failed to remove ${sdkPath(version)}, ${err}. ` +
+                `Please remove the installation for ${version} manually as it is broken.`
+        );
+    }
+};
```

Comment:
Even though the commit message on https://github.com/NordicSemiconductor/pc-nrfconnect-toolchain-manager/commit/8f38b2cc8168a9ac23c4a7ebc107939411237822 says “refactor”, replacing the `Promise.all` isn't a pure refactoring, because it changes behaviour: 

Before the two operations were done in parallel, now they are done in the sequence. And if the first fails, then the second isn't done at all. 

But it still looks fine to me and that this might be a bit slower doesn't seem so grave to me.